docs/dev/opensearch-relevancy-search.md

```diff
@@ -0,0 +1,294 @@
+
+For the sake of user experience, we decided to put on the SQL like language interface. So from the SQL language standpoint, the search functions could be performed similar to the SQL functions existing in SELECT, WHERE, GROUP BY clauses etc., and the returned data type is set to boolean. For example:
+
+```
+SELECT match(message, "this is a test") FROM my_index
+SELECT * FROM my_index where match(message, "this is a test")
+```
+
+
+Similarly from PPL standpoint, we could define it either as functions or as a command named `match`:
+**Option 1: served as eval functions similar to SQL language**
+
+```
+search source=my_index | eval f=match(message, "this is a test")
+search source=my_index | where match(message, "this is a test") | fields message
+```
+
+**Option 2: define a new command**
+
+```
+search source=my_index | match field=message query="this is a test"
+search source=my_index | match type=match_phrase field=message query="this is a test" | fields event, message, timestamp
+```
+
+### 4.2 Search mode
+The search type for the new engine is defaulted to `query_then_fetch` mode as the default setting of OpenSearch engine, and usually it works fine unless the document number is too small to smooth out the term/document frequency statistics. The alternative option for the query type is `dfs_query_then_fetch`, which adds an additional step of prequerying each shard asking about term and document frequencies before `query_then_fetch` procedures.
+
+Similarly to the DSL queries, the search mode options for SQL plugin could be supported through the request endpoint, for example to set the `dfs_query_then_fetch` as the search mode in specific query:
+```
+POST _plugins/_sql?type=dfs_query_then_fetch
+{
+  "query": "SELECT message FROM my_index WHERE match(message, 'this is a test')
+}
+```
+
+
+### 4.3 Search function details
+
+In this section we focus on defining the syntax and functionalities for each of the functions to support, and how these functions are mapped with the search engine queries. Since most of the functions are reusing the common arguments, here lists the arguments that could be used by the search functions
+
+Required parameters:
+
+* **field(s):** Field(s) you wish to search. Defaults to the index.query.default_field index setting, which has a default value of *.
+* **query:** Text, number, boolean value or date you wish to find in the provided `<field>`. For the functions that are not requiring a specific field, the `fields` parameter might be an option to specify the fields to search for terms.
+
+Optional parameters:
```

Comment:
There are much more parameters, check lists in [code](https://github.com/opensearch-project/sql/blob/main/opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/FunctionParameterRepository.java) and in [docs](https://opensearch.org/docs/latest/opensearch/query-dsl/full-text/#other-advanced-options)

Proposal: highlight available values by backticks